### PR TITLE
[Feature] Add table info

### PR DESCRIPTION
**Pull request by s7monk**

### CODE REVIEW

File: paimon-web-server/pom.xml [hunk 1]
```diff
@@ -168,6 +169,12 @@ under the License.
             <version>${hadoop.version}</version>
             <scope>provided</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.github.xiaoymin</groupId>
```
Comment (SteNicholas) [inline paimon-web-server/pom.xml:174]: Why use this dependency?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java [hunk 2]
```diff
@@ -101,13 +100,15 @@ public R<List<CatalogInfo>> getCatalog() {
...
      *
-     * @param catalogId The ID of the catalog to be removed.
-     * @return A response indicating the success or failure of the removal operation.
+     * @param catalogInfo The catalog information containing the catalog name.
+     * @return R<Void/> indicating the success or failure of the operation.
      */
-    @DeleteMapping("/{catalogId}")
-    public R<Void> remove(@PathVariable Integer catalogId) {
-        return catalogService.removeById(catalogId) ? R.succeed() : R.failed();
+    @PostMapping("/removeCatalog")
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:108]: Why not use @DeleteMapping?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 3]
```diff
@@ -66,7 +73,24 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
         try {
             Catalog catalog = CatalogUtils.getCatalog(getCatalogInfo(tableInfo.getCatalogName()));
             List<String> partitionKeys = tableInfo.getPartitionKey();
+
             Map<String, String> tableOptions = tableInfo.getTableOptions();
+            List<TableColumn> tableColumns = tableInfo.getTableColumns();
+            if (tableColumns != null && tableColumns.size() > 0) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:79]:
```suggestion
            if (!CollectionUtils.isEmpty(tableColumns)) {
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 4]
```diff
@@ -88,6 +112,276 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
...
+            }
+            TableManager.alterTable(
+                    catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), entityList);
+            if (options.size() > 0) {
+                TableManager.setOptions(
+                        catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), options);
+            }
+            return R.succeed();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:163]: Use log.error.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 5]
```diff
@@ -88,6 +112,276 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
...
+                                .columnName(tableColumn.getField())
+                                .kind(OperatorKind.DROP_COLUMN)
+                                .build();
+                entityList.add(alterTableEntity);
+            }
+            TableManager.alterTable(
+                    catalog, tableInfo.getDatabaseName(), tableInfo.getTableName(), entityList);
+            return R.succeed();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:192]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 6]
```diff
@@ -88,6 +112,276 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
...
+     * @param tableInfo The information of the table.
+     * @return The result of the operation.
+     */
+    @PostMapping("/renameColumn")
+    public R<Void> renameColumn(@RequestBody TableInfo tableInfo) {
+        try {
+            Catalog catalog = CatalogUtils.getCatalog(getCatalogInfo(tableInfo.getCatalogName()));
+            List<TableColumn> tableColumns = tableInfo.getTableColumns();
+
+            if (tableColumns.size() != 2) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:209]: Why renaming column require two table columns objects? How does the user solve this problem?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 7]
```diff
@@ -88,6 +112,276 @@ public R<Void> createTable(@RequestBody TableInfo tableInfo) {
...
+    }
+
+    /**
+     * Updates the data type of columns in a table.
+     *
+     * @param tableInfo The information of the table.
+     * @return The result of the operation.
+     */
+    @PostMapping("/updateColumnType")
+    public R<Void> updateColumnType(@RequestBody TableInfo tableInfo) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:242]: Could the updating column behavior unfied to one interface?

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@s7monk, thanks for the contribution. I left some comments for this pull request. BTW, could you add some test cases for the changes?

File: paimon-web-server/pom.xml (hunk 1 above)
Comment (s7monk) [reply paimon-web-server/pom.xml:174]: > Why use this dependency?

This provides the ability of Api Documentation for front-end development.

Comment (SteNicholas) [reply paimon-web-server/pom.xml:174]: Does this feature need this dependency?

**Comment by SteNicholas** [comment]
@s7monk, could you add the test cases for this changes?

**Pull request closed without merging**